src/components/swap/SwapButtons/SwapButtons.tsx

```diff
@@ -82,6 +115,77 @@ export function SwapButtons({
     return <ButtonConnect />
   }
 
+  if (isGnosisProtocolTradeRequireWrap) {
+    if (gnosisProtocolTradeState === GnosisProtocolTradeState.UNKNOWN)
+      setGnosisProtocolState(GnosisProtocolTradeState.WRAP)
+    const isApprovalRequired = approval !== ApprovalState.APPROVED
+
+    const isSwapDisabled =
+      !isValid ||
+      isApprovalRequired ||
+      (priceImpactSeverity > 3 && !isExpertMode) ||
+      gnosisProtocolTradeState !== GnosisProtocolTradeState.SWAP ||
+      trade === storedTrade
+
+    return (
+      <>
+        <RowBetween gap={24}>
+          {
+            <ButtonConfirmed
+              padding={'8px'}
```

Comment:
No need for curly braces.
```suggestion
              padding="8px"
```